### PR TITLE
feat: add support for ORM special columns

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@commitlint/cli": "^11.0.0",
     "@commitlint/config-conventional": "^11.0.0",
     "@map-colonies/eslint-config": "^2.2.0",
-    "@map-colonies/mc-model-types": "^11.0.1",
+    "@map-colonies/mc-model-types": "map-colonies-mc-model-types-11.0.1.tgz",
```
Comment (asafMasa) [inline package.json]: todo: will be changed

File: src/orm/generateORM.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
-import { PropertiesTypes, IOrmCatalog, PycswLayerCatalogRecord, Pycsw3DCatalogRecord, ICatalogDBEntityMapping } from '@map-colonies/mc-model-types';
+import {
+  PropertiesTypes,
+  IOrmCatalog,
+  PycswLayerCatalogRecord,
+  Pycsw3DCatalogRecord,
+  ICatalogDBEntityMapping,
+  SpecialORMColumnType,
```
Comment (alebinson) [inline src/orm/generateORM.ts]: did not see **Special**ORMColumnType
ORMColumnType is an enum

File: src/orm/generateORM.ts [hunk 3]
```diff
@@ -59,19 +66,32 @@ export class OrmGenerator {
...
+        decorators = [
+          {
+            name: name,
+            arguments: [this.objectToString(field.column as unknown as Record<string, unknown>)],
+          },
+        ];
+      } else {
+        decorators = [
+          {
+            name: 'Column',
```
Comment (alebinson) [inline src/orm/generateORM.ts]: should be taken from enum.COLUMN

File: src/orm/generateORM.ts [hunk 4]
```diff
@@ -59,19 +66,32 @@ export class OrmGenerator {
         }
         hasExclamationToken = false;
       }
+      let decorators;
+      if (field.columnType !== undefined) {
+        const name = SpecialORMColumnType[field.columnType];
+        this.importManager.addImport('typeorm', [name]);
+        decorators = [
+          {
+            name: name,
```
Comment (cl-shaharshavit-rnd) [inline src/orm/generateORM.ts]: field.columnType ?? SpecialORMColumnType.COLUMN

instead of condition

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -45,7 +45,7 @@
     "@commitlint/cli": "^11.0.0",
     "@commitlint/config-conventional": "^11.0.0",
     "@map-colonies/eslint-config": "^2.2.0",
-    "@map-colonies/mc-model-types": "^11.0.1",
+    "@map-colonies/mc-model-types": "map-colonies-mc-model-types-11.0.3.tgz",
```
Comment (asafMasa) [inline package.json]: replace version

File: src/orm/generateORM.ts (hunk 2 above)
Comment (asafMasa) [reply src/orm/generateORM.ts]: fixed

File: src/orm/generateORM.ts (hunk 3 above)
Comment (asafMasa) [reply src/orm/generateORM.ts]: fixed with shahar review

File: src/orm/generateORM.ts (hunk 4 above)
Comment (asafMasa) [reply src/orm/generateORM.ts]: fixed

**Pull request merged**